File: tasks/task12_1/point.py
import math

class Point:
    def __init__(self, x, y):
        self.x = x
        self.y = y

    def distance(self):
        return math.hypot(self.x, self.y)
    
    def distance_to(self, other):
        return math.hypot(self.x - other.x, self.y - other.y)
    

def main():
    x = float(input("Enter x coordinate: "))
    y = float(input("Enter y coordinate: "))
    
    p1 = Point(x, y)
    p2 = Point(4.5, 7.0)
    
    print(f"Distance from origin: {p1.distance()}")
    print(f"Distance to point (4.5, 7.0): {p1.distance_to(p2)}")

main()

File: tasks/task12_1/Point.kt
import kotlin.math.hypot

class Point(var x: Double, var y: Double) {
    fun distance() = hypot(x, y)

    fun distanceTo(other: Point) = hypot(x - other.x, y - other.y)
}


fun main() {
    print("Enter x coordinate: ")
    val x = readln().toDouble()
    print("Enter y coordinate: ")
    val y = readln().toDouble()

    val p1 = Point(x, y)
    val p2 = Point(4.5, 7.0)

    println("Distance from origin: ${p1.distance()}")
    println("Distance to point (4.5, 7.0): ${p1.distanceTo(p2)}")
}